### PR TITLE
Fixed OpenSSH infinite link. #2927

### LINKED ISSUE #2927: OpenSSH links into infinite loop
OpenSSH as installed via scoop has a link directory to the users home dir. Since scoop apps are in the home dir, this becomes an infinite loop.

I noticed it while trying to Windows-Search a file. It was unable to proceed the home directory. So Windows-Search is unusable.

scoop

### PR DESCRIPTION
**Pull request by excitoon**
Fixes #2927

### CODE REVIEW

File: bucket/openssh.json [hunk 1]
```diff
@@ -63,6 +63,8 @@
     "pre_install": [
         "mkdir $dir/tmp",
         "mkdir $dir/home",
-        "cmd /c mklink /j $dir\\home\\%USERNAME% %USERPROFILE%"
+        "mkdir $dir/home/$env:USERNAME",
```
Comment (Ash258) [inline bucket/openssh.json]: Use quotes around path with $dir to prevent spaces.

File: bucket/openssh.json [hunk 2]
```diff
@@ -63,6 +63,8 @@
     "pre_install": [
         "mkdir $dir/tmp",
         "mkdir $dir/home",
-        "cmd /c mklink /j $dir\\home\\%USERNAME% %USERPROFILE%"
+        "mkdir $dir/home/$env:UserName",
```
Comment (excitoon) [inline bucket/openssh.json]: Where have you been when @rasa made this PR? :)))) @Ash258

**Comment by Ash258** [comment]
Should not there be some `uninstaller.script` property, which unlink that junction?

File: bucket/openssh.json (hunk 2 above)
Comment (excitoon) [reply bucket/openssh.json]: I guess I should make it around all paths.

**Comment by excitoon** [comment]
@Ash258 I have no idea. I guess simple delete shall understand that directory junction is something to be unlinked.

**Comment by excitoon** [comment]
Personally, I think it is ready to merge. One can work on 32-bit versions for OpenSSH.
I can't properly test it because for some reason I have fresh OpenSSH in my system:
```
C:\Users\vladi\Documents\src\scoop-user>where ssh
C:\Windows\System32\OpenSSH\ssh.exe

C:\Users\vladi\Documents\src\scoop-user>ssh -V
OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
```

I guess it is Windows Update.

File: bucket/openssh.json [hunk 3]
```diff
@@ -61,8 +61,10 @@
         "usr/bin/sshd.exe"
     ],
     "pre_install": [
-        "mkdir $dir/tmp",
-        "mkdir $dir/home",
-        "cmd /c mklink /j $dir\\home\\%USERNAME% %USERPROFILE%"
+        "mkdir \"$dir/tmp\"",
+        "mkdir \"$dir/home\"",
+        "mkdir \"$dir/home/$env:UserName\"",
+        "mkdir -Force \"$env:UserProfile/.ssh\"",
```
[the remark this answers is not in the document]
Comment (excitoon) [reply bucket/openssh.json]: Because first entries do not exist and last one may already exist.

**Comment by rasa** [comment]
Should we consider instead of using msys2's ssh, we use the ssh built into git-for-windows?  See https://github.com/rasa/scoop/blob/01b832f0d9948481d7125de49191b336a1c4e6f1/bucket/openssh.json

**Comment by excitoon** [comment]
@r15ch13

**Pull request merged**